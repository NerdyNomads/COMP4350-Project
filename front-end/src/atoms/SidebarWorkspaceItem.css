.SidebarWorkspaceItem{
    width: calc(100% - 24);
    padding: 10px 0;
    padding-left: 24px;
    font-weight: 400;
    display: flex;
    flex-direction: row;
    justify-content: space-between;
    cursor: pointer;
}

.SidebarWorkspaceItem:hover{
    background-color: #F3F3F3;
}

.SidebarWorkspaceItem:hover .SidebarWorkspaceItem-edit{
    color: #7D7D7D;
    padding-right: 20px;
    visibility: visible;
}

.SidebarWorkspaceItem-edit{
    visibility: hidden;
    user-select: none;
    cursor: pointer;
}

.SidebarWorkspaceItem-selected {
    background-color: #F3F3F3;
}