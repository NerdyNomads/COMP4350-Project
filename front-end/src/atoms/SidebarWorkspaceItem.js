import React from "react";
import PropTypes from "prop-types";

import { EditIcon } from "../atoms/icons";
import {trimLongText} from "../util/util";
import "./SidebarWorkspaceItem.css";

const MAX_CHARACTER = 20;

function SidebarWorkspaceItem( {name, onEdit, selected, onClickWorkspace} ) {

  const  formattedName = trimLongText(name ,MAX_CHARACTER);

  const componentName = "SidebarWorkspaceItem";

  const handleEditWorkspaceClick = () => onEdit(true);
  

  const handleWorkspaceClick = () => onClickWorkspace(true);


  return (
    <div className={`${componentName} ${selected ? `${componentName}-selected` : "" }`} onClick={handleWorkspaceClick}>
      <div className={`${componentName}-name`}>   
        {formattedName}
      </div>
      <div className={`${componentName}-edit`} onClick={handleEditWorkspaceClick}> 
        <EditIcon/>
      </div>
    </div>
  );
}

SidebarWorkspaceItem.propTypes = {
  name: PropTypes.string.isRequired,
  onEdit: PropTypes.func.isRequired,
  onClickWorkspace: PropTypes.func.isRequired,
  selected: PropTypes.bool.isRequired
};


export default SidebarWorkspaceItem;