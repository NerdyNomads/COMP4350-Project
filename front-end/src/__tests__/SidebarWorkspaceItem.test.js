/* eslint-disable no-undef */
/* eslint-disable react/react-in-jsx-scope */
import React from "react";
import { shallow } from "enzyme";

import SidebarWorkspaceItem from "../atoms/SidebarWorkspaceItem";

test("Should match the snapshot.", () => { 
  const mockFunc = () => null;
  let wrapper = shallow(<SidebarWorkspaceItem id="" name="" selected={false} onEdit={mockFunc} onClickWorkspace={mockFunc}/>);
  expect(wrapper.html()).toMatchSnapshot();
});

test("Should call the onEdit function if the edit icon is clicked.", () => { 
  const mockFunc = () => null;
  let onEdit = jest.fn();
  let wrapper = shallow(<SidebarWorkspaceItem id="" name="" selected={false} onEdit={onEdit} onClickWorkspace={mockFunc}/>);
  
  let editDiv = wrapper.find(".SidebarWorkspaceItem-edit").first();
  editDiv.simulate("click");
  
  expect(onEdit).toBeCalledTimes(1);
});

test("Should display the whole workspace name if its length is less than 20.", () => { 
  const mockFunc = () => null;
  let workspaceName = "Short Name";
  let wrapper = shallow(<SidebarWorkspaceItem id="" name={workspaceName} selected={false} onEdit={mockFunc} onClickWorkspace={mockFunc}/>);
  
  let nameDiv = wrapper.find(".SidebarWorkspaceItem-name").first();
  expect(nameDiv.text()).toBe(workspaceName);
});

test("Should display a trimmed workspace name if its length is greater than 20.", () => { 
  const mockFunc = () => null;
  let workspaceName = "Very very very very long name";
  let wrapper = shallow(<SidebarWorkspaceItem id="" name={workspaceName} selected={false} onEdit={mockFunc} onClickWorkspace={mockFunc}/>);
  
  let nameDiv = wrapper.find(".SidebarWorkspaceItem-name").first();
  let trimmedName = "Very very very very ...";
  expect(nameDiv.text()).toBe(trimmedName);
});

test("Should call the onClickWorkspace function if the component is clicked.", () => { 
  const mockFunc = () => null;
  let onClickWorkspace = jest.fn();
  let wrapper = shallow(<SidebarWorkspaceItem id="" name="" selected={false} onEdit={mockFunc} onClickWorkspace={onClickWorkspace}/>);
  
  let sidebarWorkspaceItem = wrapper.find(".SidebarWorkspaceItem").first();
  sidebarWorkspaceItem.simulate("click");
  
  expect(onClickWorkspace).toBeCalledTimes(1);
});