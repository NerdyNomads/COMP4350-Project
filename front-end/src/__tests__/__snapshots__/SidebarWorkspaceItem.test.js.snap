// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should match the snapshot. 1`] = `"<div class=\\"SidebarWorkspaceItem \\"><div class=\\"SidebarWorkspaceItem-name\\"></div><div class=\\"SidebarWorkspaceItem-spacing\\"></div><div class=\\"SidebarWorkspaceItem-edit\\"><img alt=\\"edit\\" src=\\"/assets/images/edit.svg\\" class=\\"icon\\"/></div></div>"`;
